Deprecate ProjectReportsPluginConvention (#17056)
gradle/gradle: subprojects/docs/src/docs/userguide/core-plugins/project_report_plugin.adoc

The directory to generate the project reports into. _Default value_: `__buildDir__/__reportsDirName__`.

These convention properties are provided by a convention object of type link:{groovyDslPath}/org.gradle.api.plugins.ProjectReportsPluginConvention.html[ProjectReportsPluginConvention].

>  Configuring the project report tasks via convention properties is **deprecated**. If you need to set default values the `tasks then configure them directly. If you want to configure all report tasks of the same type in the project then use link:{javadocPath}/org/gradle/api/DomainObjectCollection.html#withType-java.lang.Class-[tasks.withType(War.class).configureEach(...)].

Reviewer: ```suggestion
 Configuring the project report tasks via the plugin's convention properties is **deprecated**. If you need to change from the default values, configure the appropriate tasks directly. If you want to configure all report tasks of the same type in the project, use link:{javadocPath}/org/gradle/api/DomainObjectCollection.html#withType-java.lang.Class-[tasks.withType(War.class).configureEach(...)].
```

Which task type is this supposed to be? Obviously it's not a War

Author: Fixed
